meta-mender-core/classes/mender-dataimg.bbclass

```diff
@@ -31,10 +32,85 @@ IMAGE_CMD_dataimg() {
     install -m 0644 "${WORKDIR}/data.${MENDER_DATA_PART_FSTYPE_TO_GEN}" "${IMGDEPLOYDIR}/${IMAGE_NAME}.dataimg"
 }
 IMAGE_CMD_dataimg_mender-image-ubi() {
+    generate_bootstrap_artifact
     mkfs.ubifs -o "${WORKDIR}/data.ubifs" -r "${IMAGE_ROOTFS}/data" ${MKUBIFS_ARGS}
     install -m 0644 "${WORKDIR}/data.ubifs" "${IMGDEPLOYDIR}/${IMAGE_NAME}.dataimg"
 }
 
+generate_bootstrap_artifact() {
+
+    if [ -z "${MENDER_ARTIFACT_NAME}" ]; then
+        bberror "Need to define MENDER_ARTIFACT_NAME variable."
+        exit 1
+    fi
+
+    if [ -z "${MENDER_DEVICE_TYPES_COMPATIBLE}" ]; then
+        bbfatal "MENDER_DEVICE_TYPES_COMPATIBLE variable cannot be empty."
+    fi
+
+    extra_args=
+
+    for dev in ${MENDER_DEVICE_TYPES_COMPATIBLE}; do
+        extra_args="$extra_args -t $dev"
+    done
+
+    if [ -n "${MENDER_ARTIFACT_SIGNING_KEY}" ]; then
+        extra_args="$extra_args -k ${MENDER_ARTIFACT_SIGNING_KEY}"
+    fi
+
+    if [ -n "${MENDER_ARTIFACT_NAME_DEPENDS}" ]; then
+        cmd=""
+        apply_arguments "--artifact-name-depends" "${MENDER_ARTIFACT_NAME_DEPENDS}"
+        extra_args="$extra_args  $cmd"
+    fi
+
+    if [ -n "${MENDER_ARTIFACT_PROVIDES}" ]; then
+        cmd=""
+        apply_arguments "--provides" "${MENDER_ARTIFACT_PROVIDES}"
+        extra_args="$extra_args  $cmd"
+    fi
+
+    if [ -n "${MENDER_ARTIFACT_PROVIDES_GROUP}" ]; then
+        cmd=""
+        apply_arguments "--provides-group" "${MENDER_ARTIFACT_PROVIDES_GROUP}"
+        extra_args="$extra_args $cmd"
+    fi
+
+    if [ -n "${MENDER_ARTIFACT_DEPENDS}" ]; then
+        cmd=""
+        apply_arguments "--depends" "${MENDER_ARTIFACT_DEPENDS}"
+        extra_args="$extra_args $cmd"
+    fi
+
+    if [ -n "${MENDER_ARTIFACT_DEPENDS_GROUPS}" ]; then
+        cmd=""
+        apply_arguments "--depends-groups" "${MENDER_ARTIFACT_DEPENDS_GROUPS}"
+        extra_args="$extra_args $cmd"
+    fi
+
+    img_checksum="$(sha256sum --binary ${IMGDEPLOYDIR}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.${ARTIFACTIMG_FSTYPE} | cut --delimiter=' ' --fields=1)"
+
+    if [ -z "${img_checksum}" ]; then
+        bberror "The image checksum cannot be empty"
+    fi
+
+    mender-artifact write bootstrap-artifact \
+        --artifact-name ${MENDER_ARTIFACT_NAME} \
+        --provides "rootfs-image.version:${MENDER_ARTIFACT_NAME}" \
+        --provides "rootfs-image.checksum:${img_checksum}" \
```

Comment:
Missing:

```
--clears-provides "rootfs-image.*"
```

Reply:
It's probably smart to include it, yes. When the bootstrap artifact is used as we intend it to, then this flag is guaranteed to be redundant, because the database must be empty for this artifact to be installed. But it is possible for someone to take the same artifact and use it for a manual installation, in which case it would be relevant.

If we add it here, then we probably should in mender-convert as well.

Reply:
As I see it, it is the same as with signing the Artifact. It is redundant for the intended use, but we keep it for other use cases like you describe.

> If we add it here, then we probably should in mender-convert as well.

I agree.

I am sorry @oleorhagen that I didn't catch up this in the other PR :crying_cat_face:

Reply:
Why, we haven't released squat, it's still even in feature branches :)

We can do whatever we want :muscle:

Reply:
Fixing btw

Reply:
convert-pr: https://github.com/mendersoftware/mender-convert/pull/471